### ISSUE TITLE
Error downloading IERS A file

### ISSUE THREAD

**Issue by paulray** [astropy/astropy#9499]
### Description

Downloading the IERS A file fails with an HTTP error, even on the new 3.2.3 version.

### Steps to Reproduce

```python
python -c "import astropy; print(astropy.__version__); from astropy.utils.data import download_file; from astropy.utils.iers import IERS_A_URL; print(IERS_A_URL); download_file(IERS_A_URL, cache=True)"
3.2.3
https://datacenter.iers.org/data/9/finals2000A.all
Traceback (most recent call last):
  File "<string>", line 1, in <module>
  File "/Users/paulr/Library/Python/3.6/lib/python/site-packages/astropy/utils/data.py", line 1105, in download_file
    raise e
  File "/Users/paulr/Library/Python/3.6/lib/python/site-packages/astropy/utils/data.py", line 1024, in download_file
    with urllib.request.urlopen(remote_url, timeout=timeout) as remote:
  File "/opt/local/Library/Frameworks/Python.framework/Versions/3.6/lib/python3.6/urllib/request.py", line 223, in urlopen
    return opener.open(url, data, timeout)
  File "/opt/local/Library/Frameworks/Python.framework/Versions/3.6/lib/python3.6/urllib/request.py", line 532, in open
    response = meth(req, response)
  File "/opt/local/Library/Frameworks/Python.framework/Versions/3.6/lib/python3.6/urllib/request.py", line 642, in http_response
    'http', request, response, code, msg, hdrs)
  File "/opt/local/Library/Frameworks/Python.framework/Versions/3.6/lib/python3.6/urllib/request.py", line 570, in error
    return self._call_chain(*args)
  File "/opt/local/Library/Frameworks/Python.framework/Versions/3.6/lib/python3.6/urllib/request.py", line 504, in _call_chain
    result = func(*args)
  File "/opt/local/Library/Frameworks/Python.framework/Versions/3.6/lib/python3.6/urllib/request.py", line 650, in http_error_default
    raise HTTPError(req.full_url, code, msg, hdrs, fp)
urllib.error.HTTPError: HTTP Error 403: Forbidden
```

### System Details

```python
mcxr2 : 25>python
Python 3.6.9 (default, Sep  7 2019, 20:24:16) 
[GCC 4.2.1 Compatible Apple LLVM 8.0.0 (clang-800.0.42.1)] on darwin
Type "help", "copyright", "credits" or "license" for more information.
>>> import platform; print(platform.platform())
Darwin-15.6.0-x86_64-i386-64bit
>>> import sys; print("Python", sys.version)
Python 3.6.9 (default, Sep  7 2019, 20:24:16) 
[GCC 4.2.1 Compatible Apple LLVM 8.0.0 (clang-800.0.42.1)]
>>> import numpy; print("Numpy", numpy.__version__)
Numpy 1.17.2
>>> import scipy; print("Scipy", scipy.__version__)
Scipy 1.3.1
>>> import astropy; print("astropy", astropy.__version__)
astropy 3.2.3
```

**Comment by pllim** [comment]
Did you try using the mirror (FTP)?

**Comment by paulray** [comment]
No, I don't know how to tell it to use the mirror.

I did verify that I could download the file with that URL using either curl or my browser, so it seems like the `download_file()` function may not be doing something right?

**Comment by pllim** [comment]
Mirror is at
```python
from astropy.utils.iers import IERS_A_URL_MIRROR
```

**Comment by pllim** [comment]
@eteq swapped them out in #9487 , so if the mirror works for you, then this will be a non-issue in the next release.

**Comment by paulray** [comment]
Indeed the mirror works fine!

**Comment by pllim** [comment]
This is still a problem though, as I can also reproduce this. If everyone sees this behavior, then we basically won't have a mirror.

**Comment by pllim** [comment]
@taldcroft , @mhvk , @eteq -- Is this an issue caused by service provider configuration for https://datacenter.iers.org/data/9/finals2000A.all ? I tried inserting a "User-Agent" but that didn't work either. My variations either resulted in Error 403 (forbidden) or 0 (???).

Example with User-Agent:
```python
import urllib
r = urllib.request.Request(IERS_A_URL, headers={'User-Agent': 'astropy/iers'})
u = urllib.request.urlopen(r)
```

**Comment by StuartLittlefair** [comment]
I can confirm that I'm seeing this behaviour too. By contrast the code below works just fine..

```
import requests
r = requests.get(IERS_A_URL)
```

as does

```
import urllib
r = urllib.request.Request(IERS_A_URL, headers={'User-Agent': 'astropy/iers', 'Accept': '*/*'}) 
u = urllib.request.urlopen(r)
```

**Comment by pllim** [comment]
So, we might have to allow custom headers to be passed into `download_file`... 🤔 

Let's wait until after #9182 is merged.

**Comment by dr-rodriguez** [comment]
We're also seeing similar issues with IERS-B that may be related:
```
from astropy import __version__
print(__version__)
3.1.1
from astropy.time import Time
print(Time.now().ut1.to_datetime())
WARNING: failed to download http://maia.usno.navy.mil/ser7/finals2000A.all and http://toshi.nofs.navy.mil/ser7/finals2000A.all, using local IERS-B: [Errno 54] Connection reset by peer;HTTP Error 503: Service Unavailable [astropy.utils.iers.iers]
```

This is used by some of our operational processing scripts at MAST to load TESS observations.

**Comment by pllim** [comment]
USNO ones are because of #9427 ; there is nothing we can do about that one.

**Comment by dr-rodriguez** [comment]
Ah, I had missed that. Thanks!

**Comment by eteq** [comment]
@pllim - right, but it's possible @dr-rodriguez is facing a different issue, since it's IERS B, which potentailly *was* fixed in #9427 . I was just pointing out that this should be made a separate issue if it's distinct from this one.

**Comment by eteq** [comment]
Back to the main issue: I agree with @pllim we should wait until after #9182 is merged since that would otherwise be a challenging rebase.  But I will start a PR with some test to reproduce this and see if I can hack a quick fix (for merging after #9182)
